test/deprecates.jl

```diff
@@ -0,0 +1,201 @@
+using StatsBase
+using Base.Test
+using Compat
+import Compat: view
+
+@testset "StatsBase.Deprecates" begin
+
+@testset "Deprecates WeightVec and weights" begin
```

Comment:
Okay, I'll add an item to my checklist to remove it before dropping the "[WIP]" from this PR, but please just ignore that until then.